### PR TITLE
feat: param based redirection

### PR DESCRIPTION
**Pull request by mubbsharanwar**
Register success redirection would be based on the next query params.

VAN-1535

### CODE REVIEW

File: src/progressive-profiling/ProgressiveProfiling.jsx [hunk 1]
```diff
@@ -171,6 +172,9 @@ const ProgressiveProfiling = (props) => {
 
   const handleSkip = (e) => {
     e.preventDefault();
+    if (nextUrl) {
+      window.location = nextUrl;
```
Comment (zainab-amir) [inline src/progressive-profiling/ProgressiveProfiling.jsx:176]: why do we need to add it here?

**Review by zainab-amir — COMMENTED** [review]
Why are we not using existing logic in use effect where we set the optional fields?

File: src/progressive-profiling/ProgressiveProfiling.jsx (hunk 1 above)
Comment (mubbsharanwar) [reply src/progressive-profiling/ProgressiveProfiling.jsx:176]: We also need to redirect to the next URL for the skip button click.

File: src/common-components/data/reducers.js [hunk 2]
```diff
@@ -3,7 +3,10 @@ import { COMPLETE_STATE, FAILURE_STATE, PENDING_STATE } from '../../data/constan
 
 export const defaultState = {
   fieldDescriptions: {},
-  optionalFields: {},
+  optionalFields: {
+    fields: [],
+    extended_profile: [],
+  },
```
Comment (zainab-amir) [inline src/common-components/data/reducers.js:9]: Why have we changed the default state?

File: src/progressive-profiling/ProgressiveProfiling.jsx [hunk 3]
```diff
@@ -90,17 +91,18 @@ const ProgressiveProfiling = (props) => {
 
   useEffect(() => {
     if (registrationEmbedded) {
-      getFieldDataFromBackend({ is_welcome_page: true });
+      getFieldDataFromBackend({ is_welcome_page: true, next: queryParams?.next });
     }
-  }, [registrationEmbedded, getFieldDataFromBackend]);
+  }, [registrationEmbedded, getFieldDataFromBackend, queryParams?.next]);
```
Comment (zainab-amir) [inline src/progressive-profiling/ProgressiveProfiling.jsx:96]: We can use queryParams directly
```suggestion
  }, [registrationEmbedded, getFieldDataFromBackend, queryParams]);
```

File: src/progressive-profiling/ProgressiveProfiling.jsx [hunk 4]
```diff
@@ -172,7 +172,9 @@ const ProgressiveProfiling = (props) => {
   const handleSkip = (e) => {
     e.preventDefault();
     window.history.replaceState(props.location.state, null, '');
-    setShowModal(true);
+    if (welcomePageContext.nextUrl !== '') {
+      setShowModal(true);
+    }
```
Comment (zainab-amir) [inline src/progressive-profiling/ProgressiveProfiling.jsx:177]: Why are we changing the functionality?

Comment (mubbsharanwar) [reply src/progressive-profiling/ProgressiveProfiling.jsx:177]: Modal shows the button, click to redirect to the dashboard but in the embedded experience we want to redirect to the next URL.

File: src/progressive-profiling/ProgressiveProfiling.jsx (hunk 3 above)
Comment (mubbsharanwar) [reply src/progressive-profiling/ProgressiveProfiling.jsx:96]: We can't use objects directly in dependency. It causes the infinite useEffect loop due to memory address change.

Comment (syedsajjadkazmii) [reply src/progressive-profiling/ProgressiveProfiling.jsx:96]: @mubbsharanwar the useEffect should not run infinite when having object as dependency, it must be some other reason that useEffect is running infinitely for queryParams.

Comment (zainab-amir) [reply src/progressive-profiling/ProgressiveProfiling.jsx:96]: Yes. The queryParams are constant and will only change once

File: src/progressive-profiling/ProgressiveProfiling.jsx (hunk 4 above)
Comment (zainab-amir) [reply src/progressive-profiling/ProgressiveProfiling.jsx:177]: Then lets change the text of the button. We don't want to change functionality without asking product and UX

File: src/progressive-profiling/ProgressiveProfiling.jsx (hunk 3 above)
Comment (mubbsharanwar) [reply src/progressive-profiling/ProgressiveProfiling.jsx:96]: Why should not we use objects as a dependency? https://dmitripavlutin.com/react-useeffect-infinite-loop/#21-avoid-objects-as-dependencies

File: src/progressive-profiling/ProgressiveProfiling.jsx (hunk 4 above)
Comment (mubbsharanwar) [reply src/progressive-profiling/ProgressiveProfiling.jsx]: Actually, I didn't set the env variables locally that's why I was getting the label button on the modal button "Go to dashboard". On stage and prod the modal button label is "Go to edX" which is fine. We don't need to add this check.

**Review by zainab-amir — APPROVED** [review]

**Pull request merged**